content/blog/summary-from-the-9th-annual-chapel-implementers-and-users-workshop-chiuw-2022.md

```diff
@@ -0,0 +1,73 @@
+
+**Chapel Project Updates**
+
+One of the updates from the Chapel developers was that the new parser was being used in the production Chapel compiler.  This is important because an often-heard complaint from Chapel users is that the compiler is too slow.  The current production compiler does whole program compilation and thus is not able to take advantage of separate and incremental compilation approaches.  The current *dyno* effort within the Chapel team is redesigning the Chapel compiler to enable separate, incremental, and in general more dynamic compilation.
+
+Another important update for Chapel is the ever-growing GPU support.  Currently some `forall` loops in Chapel are compiled for CPUs and as GPU kernels.  Which version to run is selected at runtime.  Below is an example of Chapel code that currently runs as a GPU kernel on machines where one or more GPUs are available.
+
+![coforall gpu in here.gpus do on gpu { var A, B, C: [1..n] real; const alpha = 2.0; B = 1.0; C = 2.0; A = B + alpha + C; }](/img/chiuw22-gpu-code-example.jpg "Chapel Code that is offloaded to all the gpus on a locale/node.")
+
+Keep an eye on the [GPU Programming Technical Note](https://chapel-lang.org/docs/technotes/gpu.html) for new features as GPU support in Chapel continues to expand.  More GPU support means handling an especially difficult kind of parallelism that programmers struggle with these days.
+
+**Feedback from Attendees**
+
+One user, Nelson Dias, gave a talk about moving to Chapel from the C and Python programming languages.  Nelson’s abstract states, “Chapel is a very elegant language, providing the power and speed of C and Fortran, while allowing a high degree of abstraction and expressiveness that rivals Python's. I have used it in the last two years for: calculating statistics over massive turbulence datasets, implementing models for lake evaporation in hydrology, and testing some relatively simple numerical solutions of partial differential equations.”  [His talk](https://youtu.be/400jmMzdzHQ) details the advantages and disadvantages he found while programming in Chapel.
+
+In a post-workshop survey, attendees provided the following feedback,
+
+* Ways to improve CHIUW: "maybe some short tutorials on how to use language features in a real example. Also debugging and performance analysis tutorial."  The Chapel team plans on doing these during next year’s Chapel Coding Day.
+* Some of the suggestions were requests for topics in presentations.  The CHIUW organizers can encourage such submissions from the community next year.
+* One person wants to see how people are doing performance optimizations in Chapel applications.  For next year, we will encourage such submissions from the community.
+* Two people mentioned wanting to hear about libraries more.  One about parallel/distributed libraries in Chapel and the other about wrapping C libraries in Chapel and lightweight Python wrappers for Chapel.
```

Comment:
"...wanting to hear more about libraries."

Reply:
done